BootloaderCommonPkg/Library/ContainerLib/ContainerLib.c

```diff
@@ -543,16 +543,18 @@ LocateComponentEntry (
 
   // Locate the component from the container header
   ContainerHdr = (CONTAINER_HDR *)(UINTN)ContainerEntry->HeaderCache;
-  CompEntry = LocateComponentEntryFromContainer (ContainerHdr, ComponentName);
-  if (CompEntry == NULL) {
-    return EFI_NOT_FOUND;
+  if (ComponentName != 0) {
+    CompEntry = LocateComponentEntryFromContainer (ContainerHdr, ComponentName);
+    if (CompEntry == NULL) {
+      return EFI_NOT_FOUND;
+    }
   }
 
   if (ContainerEntryPtr != NULL) {
     *ContainerEntryPtr = ContainerEntry;
   }
 
-  if (ComponentEntryPtr != NULL) {
+  if ((ComponentEntryPtr != NULL) && (ComponentName != 0)) {

```

Comment:
I don't think we need to change this.

Reply:
when component name is 0,  CompEntry will have NULL.  It is fine to write NULL back into ComponentEntryPtr.